### ISSUE TITLE
calling headers.add gives a TypeError

### ISSUE THREAD

**Issue by grundid** [pulyaevskiy/node-interop#26]
I'm not sure if this is the right repo. I'm using Dart with Cloud Functions.

When I call headers.add on a response object I get the following error. Calling headers.set is working.
Since I haven't called headers.set prior to headers.add I assume that headers are not initialized here.

```
TypeError: receiver.getHeaders is not a function
    at UnknownJavaScriptObject.getHeaders$0 (/user_code/build/dart/index.dart.js:823:25)
    at J.getHeaders$0$x (/user_code/build/dart/index.dart.js:9526:41)
    at ResponseHttpHeaders.get$nativeHeaders (/user_code/build/dart/index.dart.js:8694:18)
    at ResponseHttpHeaders.$index (/user_code/build/dart/index.dart.js:8715:22)
    at ResponseHttpHeaders.add$2 (/user_code/build/dart/index.dart.js:8737:47)
    at /user_code/build/dart/index.dart.js:8975:25
```

**Comment by pulyaevskiy** [comment]
Thanks for the report!

Looks like it could be because `getHeaders()` was added in Node.js 7.7.0 and GCP Cloud Functions run on 6.14.0...

I'll try and see if this can be worked around somehow.

**Comment by alextekartik** [comment]
As I did encounter the same think, I created a quick workaround in a fork here:
https://github.com/tekartik/node-interop/commit/b8df60ab7b1e93a01baa374d7ff0c25ce3a74035

It does work on Node pre 7.7.0. In my solution I added a runtime check of the current node version, not sure if that's the best solution. I was able to deploy my firebase cloud function and set the headers in my response but I did not do much more testing on it...it is wrong in many (calling add on the same key won't work) ways but since the only thing I do is adding single value headers, that was enough.
